mlibrary/my_account
# Horizontal Navigation

# Overview
As a part of getting ILLiad up and running on the site, the `From Other Institutions (Interlibrary Loan)` page needed to be added under `Requests`. This required horizontal navigation to be added (similar to what you see in `Shelf`), redirects to be made, and unit tests to be updated.

## Horizontal Navigation
In an effort to keep programming DRY, I created the `horizontal_nav` view and model to help generate the secondary navigations for all pages that are under `Shelf` and `Requests`. Any changes that need to be made in that area can be done within that file.

## Testing
- Run `docker-compose run web bundle exec rspec` to make sure the tests pass. Try and break the tests to double-check.
- Run `npm run compile` to make sure the styles compile successfully.
- Check if [http://localhost:4567/requests/](http://localhost:4567/requests/) redirects you to `http://localhost:4567/requests/um-library`.
- Check all `Shelf` and `Requests` pages to make sure nothing is broken.
